ui/app/AppLayouts/Chat/ChatColumn/ChatComponents/SignTransactionModal.qml

```diff
@@ -0,0 +1,190 @@
+
+ModalPopup {
+    property var selectedAccount
+    property var selectedRecipient
+    property var selectedAsset
+    property var selectedAmount
+    property var selectedFiatAmount
+    property var selectedGasLimit
+    property var selectedGasPrice
+
+    id: root
+
+    //% "Send"
+    title: qsTrId("command-button-send")
+    height: 504
+
+    property MessageDialog sendingError: MessageDialog {
+        id: sendingError
+        title: qsTr("Error sending the transaction")
+        icon: StandardIcon.Critical
+        standardButtons: StandardButton.Ok
+    }
+    property MessageDialog sendingSuccess: MessageDialog {
+        id: sendingSuccess
+        //% "Success sending the transaction"
+        title: qsTrId("success-sending-the-transaction")
+        icon: StandardIcon.NoIcon
+        standardButtons: StandardButton.Ok
+        onAccepted: {
+            root.close()
+        }
+    }
+
+    onClosed: {
+        stack.reset()
+    }
+
+    function sendTransaction() {
+        let responseStr = walletModel.sendTransaction(root.selectedAccount.address,
+                                                 root.selectedRecipient.address,
+                                                 root.selectedAsset.address,
+                                                 root.selectedAmount,
+                                                 gasSelector.selectedGasLimit,
+                                                 gasSelector.selectedGasPrice,
+                                                 transactionSigner.enteredPassword)
+        let response = JSON.parse(responseStr)
+
+        if (response.error) {
+            if (response.error.includes("could not decrypt key with given password")){
+                transactionSigner.validationError = qsTr("Wrong password")
+                return
+            }
+            sendingError.text = response.error
+            return sendingError.open()
+        }
+
+        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2/%1").arg(response.result).arg(walletModel.etherscanLink)
+        sendingSuccess.open()
+    }
+
+    TransactionStackView {
+        id: stack
+        anchors.fill: parent
+        anchors.leftMargin: Style.current.padding
+        anchors.rightMargin: Style.current.padding
+        onGroupActivated: {
+            root.title = group.headerText
+            btnNext.label = group.footerText
+        }
+        TransactionFormGroup {
+            id: group1
+            headerText: qsTr("Send")
+            footerText: qsTr("Preview")
+
+            GasSelector {
+                id: gasSelector
+                anchors.topMargin: Style.current.bigPadding
+                slowestGasPrice: parseFloat(walletModel.safeLowGasPrice)
+                fastestGasPrice: parseFloat(walletModel.fastestGasPrice)
+                getGasEthValue: walletModel.getGasEthValue
+                getFiatValue: walletModel.getFiatValue
```

Comment:
Still working on the PR (will be up tomorrow), but this is how I have implemented the gas estimation from the qml side:
https://github.com/status-im/nim-status-client/blob/735b59a443a7f3f15474c7c50b78883d31244769/ui/app/AppLayouts/Wallet/SendModal.qml#L146-L163

Reply:
Ok, let me know when you have the simplified version of it

Reply:
It wasn't not really simplified much from the above on the QML side, but was simplified a lot on the nim side. Anyway, you can put the gas estimate in like so:

https://github.com/status-im/nim-status-client/blob/735b59a443a7f3f15474c7c50b78883d31244769/ui/app/AppLayouts/Wallet/SendModal.qml#L146-L163

Then replace:
1. `txtAmount.selectedAsset.address` with `JSON.parse(walletModel.getStatusToken()).address`
2. `txtAmount.selectedAmount` with the amount being sent

You can see it all put together in https://github.com/status-im/nim-status-client/pull/868/files#diff-735aae29e4b17fa24782e30732a3befe, however the `gasEstimate` function is a bit different because that's a sticker purchase. Whereas, the estimation for you is a standard token tx.